pkg/cmd/exoscale.go

```diff
@@ -84,22 +83,29 @@ func ExoscaleCmds() *cli.Command {
 				Usage:  "Get metrics from object storage service",
 				Before: addCommandName,
 				Action: func(c *cli.Context) error {
-					log := log.AppLogger(c.Context)
-					ctrl.SetLogger(log)
+					logger := log.Logger(c.Context)
+					ctrl.SetLogger(logger)
 
-					log.Info("Creating Exoscale client")
-					exoscaleClient, err := exoscale.InitClient(accessKey, secret)
+					logger.Info("Creating Exoscale client")
+					exoscaleClient, err := exoscale.NewClient(accessKey, secret)
 					if err != nil {
 						return fmt.Errorf("exoscale client: %w", err)
 					}
 
-					log.Info("Creating k8s client")
-					k8sClient, err := cluster.NewClient(kubeconfig, k8sServerURL, k8sServerToken)
+					logger.Info("Creating k8s client")
+					k8sClient, err := kubernetes.NewClient(kubeconfig, kubernetesServerURL, kubernetesServerToken)
 					if err != nil {
 						return fmt.Errorf("k8s client: %w", err)
 					}
 
-					o := sos.NewObjectStorage(exoscaleClient, k8sClient, dbURL)
+					now := time.Now().In(time.UTC)
+					previousDay := now.Day() - 1
+					billingDate := time.Date(now.Year(), now.Month(), previousDay, billingHour, 0, 0, 0, now.Location())
```

Comment:
not sure if this will work with the first day of the month (`previousDay` calculation)

Reply:
[`time.Date()`](https://pkg.go.dev/time#Date) handles this and other cases well, see: https://go.dev/play/p/Z0k2i8n8Cis